### PR TITLE
I/O exercise solution

### PR DESCRIPTION
**Pull request by Genyus**
Ignore additional directories incuded due to rebase, relevant changes are in input-output folder

### CODE REVIEW

File: exercises/input-output/README.md [hunk 1]
```diff
@@ -12,22 +12,47 @@ For the exercises below, we've provided the starter project above.
...
 
-The output file should be created in this project's resources folder.
+```java
+public void copy(String inputFile, String outputFile)
+```
+
+The `copy` method should:
+- copy the contents of `inputFile` to `outputFile`.
+- create `outputFile` if it doesn't exist and overwrite it, if it does
+- look in this project's resources folder for both `inputFile` and `outputFile`
```
Comment (tanyapowell) [inline exercises/input-output/README.md]: What is the file type(s)?

**Review by tanyapowell — COMMENTED** [review]
It should be made clearer in the README.md that the provided `inputFile` is the [exercise.txt](https://github.com/cbfacademy-solutions/java-course-exercises/blob/main/exercises/input-output/src/main/resources/exercise.txt) and perhaps add an example of what the output could be

**Review by sohbaker — COMMENTED** [review]
Code LGTM. Agree with the queries around the README

File: exercises/input-output/README.md (hunk 1 above)
Comment (Genyus) [reply exercises/input-output/README.md]: Plaintext

File: exercises/input-output/README.md [hunk 2]
```diff
@@ -12,22 +12,49 @@ For the exercises below, we've provided the starter project above.
...
-The output file should be created in this project's resources folder.
+### :bookmark: Note:
+The paths to `inputFile` and `outputFile` are relative to the root directory of this repository (e.g. `exercises/input-output/src/main/resources/exercise.txt`)
 
 ## :pushpin: `java.nio.*`
 
-Using the `java.nio.*` classes, improve on the previous program.
+Using the `java.nio.*` classes, improve the `FileManager` class:
 
 ## :pushpin: File Content Handling
```
Comment (tanyapowell) [inline exercises/input-output/README.md]: Either remove the `:` at the end of line 41 or make line 43 a subheading of the `java.nio.*` section. It's reading as though `## :pushpin: File Content Handling` is a subtask of the `java.nio.*` task

**Review by tanyapowell — CHANGES_REQUESTED** [review]

**Review by sohbaker — APPROVED** [review]
The changes specific to I/O look good. There are files from other exercises on this PR though (algorithms, oop and opp homework)

**Review by sohbaker — CHANGES_REQUESTED** [review]
Meant to request changes

File: exercises/input-output/README.md (hunk 2 above)
Comment (Genyus) [reply exercises/input-output/README.md]: Updated content

File: exercises/input-output/src/main/java/com/cbfacademy/FileManager.java [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardCopyOption;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * A utility class for managing file operations.
+ */
+public class FileManager {
```
Comment (sohbaker) [inline exercises/input-output/src/main/java/com/cbfacademy/FileManager.java:20]: Can we make it clearer in this class that there are two steps to some of these exercises, in the way that it's laid out? I'm thinking of Instructors etc who'll come to this completely fresh and making it simpler to follow what's happening.

e.g. 
```
    /**
     * Step One: Read a text file using a BufferedReader and return a Stream of lines.
     * Step Two: Update this method to read a file using the nio class
     * 
     * @param filename The name of the file to read.
     * @return A Stream of lines from the file.
     * @throws IOException If an I/O error occurs while reading the file.
     */
    public List<String> readFile(String filename) throws IOException {
        // Step One Solution
        // try (BufferedReader reader = new BufferedReader(new
        // FileReader(filePath.toString()))) {
        // return reader.lines().collect(Collectors.toList());
        // }

        // Step Two Solution
        Path filePath = Paths.get(filename);

        return Files.readAllLines(filePath);
    }
```

**Review by sohbaker — CHANGES_REQUESTED** [review]
Made a comment on the I/O solution file. Outside of that the rest of the code for I/O looks good. 

There are still a number of files from other exercises on here too though.

File: exercises/input-output/src/main/java/com/cbfacademy/FileManager.java (hunk 3 above)
Comment (Genyus) [reply exercises/input-output/src/main/java/com/cbfacademy/FileManager.java:20]: The other files got added due to a rebase and I really can't be bothered to figure out how to remove them. I'll amend the solution comments as suggested

File: exercises/input-output/README.md [hunk 4]
```diff
@@ -5,37 +5,107 @@
...
-We've provided the input `exercise.txt` file in this project's resources folder.
+- `public void reverseLines(String inputFile, String outputFile)`
+  - read the contents of `inputFile`
+  - write the lines to `outputFile`, but in reverse order; the first line will be the last one, the second will be the penultimate, and so on, until the last line of the original file, which should appear in the first position of the resulting file.
+  - create `outputFile` if it doesn't exist and overwrite it, if it does
+  - look in this project's resources folder for both `inputFile` and `outputFile`
+  - not reuse any of the previous methods
 
-Write code that reads the content of the `exercise.txt`, then writes them to an `inverted.txt` file. 
+### Part 4
```
Comment (sohbaker) [inline exercises/input-output/README.md]: Part 4 looks like a duplication of Part 3.

**Review by sohbaker — CHANGES_REQUESTED** [review]
One change to the README needed.

**Review by sohbaker — APPROVED** [review]
LGTM

**Pull request merged**